Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,408 @@
+    }
+    if let server = self.server {
+      XCTAssertNoThrow(try server.close().wait())
+    }
+
+    super.tearDown()
+  }
+
+  func testFileByFileName() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.fileByFilename = "bar1.proto"
+      }
+    )
+    serviceReflectionInfo.requestStream.finish()
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(
+        serializedData: (message?.fileDescriptorResponse
+          .fileDescriptorProto[0])!
+      )
+    XCTAssertEqual(receivedFileDescriptorProto.name, "bar1.proto")
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.count,
+      1
+    )
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.first!.method.first!.name,
+      "testMethod1"
+    )
+    XCTAssertEqual(message?.fileDescriptorResponse.fileDescriptorProto.count, 4)
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.listServices = "services"
+      }
+    )
+
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }.sorted()
+    XCTAssertEqual(receivedServices?.count, 4)
+
+    let servicesNames = self.getServicesNamesFromProtos(
+      protos: self.makeProtosWithDependencies()
+    ).sorted()
+
+    for index in 0 ... servicesNames.count - 1 {
+      XCTAssertEqual(receivedServices![index], servicesNames[index])
+    }
+  }
+
+  func testReflectionServiceDataFileDescriptorDataByFilename() throws {
+    var protos = self.makeProtosWithDependencies()
+    let registry = try ReflectionServiceData(fileDescriptors: protos)
+
+    let registryFileDescriptorData = registry.fileDescriptorDataByFilename
+
+    for (fileName, protoData) in registryFileDescriptorData {
+      let serializedFiledescriptorData = protoData.serializedFileDescriptorProto
+      let dependencyFileNames = protoData.dependencyFileNames
+
+      guard let index = protos.firstIndex(where: { $0.name == fileName }) else {
+        return XCTFail(
+          """
+          Could not find the file descriptor proto of \(fileName)
+          in the original file descriptor protos list.
```

Comment:
Best to keep messages as a single lines.

The `"""` format will put new lines where you've put (i.e. this will be split over two lines). We should leave splitting the text over new lines to whatever tool (e.g. Xcode) is running the test.

We can avoid the newlines here by doing:

```suggestion
          Could not find the file descriptor proto of \(fileName) \
          in the original file descriptor protos list.
```